### PR TITLE
Feature/memoapp

### PR DESCRIPTION
**Pull request by pusztaienike**

### CODE REVIEW

File: examples/sn-react-memoapp/package.json [hunk 1]
```diff
@@ -0,0 +1,57 @@
+{
+  "name": "@sensenet/react-memoapp",
+  "version": "1.0.2",
+  "private": true,
+  "author": "Sense/Net <snteam@sensenet.com> (http://sensenet.com/)",
+  "contributors": [
+    "Dániel Blaskó <blaskoi@gmail.com>",
+    "Zoltán Bedi <zoltan.bedi@sensenet.com>"
```
Comment (B3zo0) [inline examples/sn-react-memoapp/package.json]: Please remove me from contributors.

File: examples/sn-react-memoapp/package.json [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    "build": "react-scripts build",
+    "eject": "react-scripts eject"
+  },
+  "devDependencies": {
+    "@types/jest": "^24.0.23",
+    "@types/node": "^12.12.14",
+    "@types/react": "^16.9.13",
+    "@types/react-dom": "^16.9.4",
+    "react": "^16.12.0",
+    "react-dom": "^16.12.0",
```
Comment (B3zo0) [inline examples/sn-react-memoapp/package.json]: Please remove duplicate react, react-dom packages.

File: examples/sn-react-memoapp/public/index.html [hunk 3]
```diff
@@ -0,0 +1,15 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <meta name="theme-color" content="#000000" />
+    <meta name="description" content="Sensenet starter created using create-react-app" />
```
Comment (B3zo0) [inline examples/sn-react-memoapp/public/index.html]:
```suggestion
    <meta name="description" content="sensenet starter created using create-react-app" />
```

File: examples/sn-react-memoapp/public/index.html [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    <title>Sensenet starter</title>
```
Comment (B3zo0) [inline examples/sn-react-memoapp/public/index.html]:
```suggestion
    <title>sensenet starter</title>
```

File: examples/sn-react-memoapp/test/repository-provider.test.tsx [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import { mount } from 'enzyme'
+import React from 'react'
+import { act } from 'react-dom/test-utils'
+import { RepositoryProvider } from '../src/context/repository-provider'
+
+describe('The repository provider instance', () => {
+  it('should renders correctly', async () => {
+    let wrapper: any
+    await act(async () => {
```
Comment (B3zo0) [inline examples/sn-react-memoapp/test/repository-provider.test.tsx]: Please remove this file.

File: examples/sn-react-memoapp/public/index.html [hunk 6]
```diff
@@ -0,0 +1,15 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <meta name="theme-color" content="#000000" />
+    <meta name="description" content="sensenet starter created using create-react-app" />
+    <title>sensenet starter</title>
```
Comment (herflis) [inline examples/sn-react-memoapp/public/index.html]:
```suggestion
    <title>sensenet memo app example</title>
```

File: examples/sn-react-memoapp/test/add-new-memo.test.tsx [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+import { AddNew } from '../src/components/add-new-memo'
+
+describe('The new memo panel instance', () => {
+  const addnewprops = {
+    show: true,
+    onCreate: jest.fn(),
+    onClose: jest.fn(),
+  }
+
+  it('should renders correctly', () => {
```
Comment (herflis) [inline examples/sn-react-memoapp/test/add-new-memo.test.tsx]:
```suggestion
  it('should be rendered correctly', () => {
```

File: examples/sn-react-memoapp/test/app.test.tsx [hunk 8]
```diff
@@ -0,0 +1,9 @@
+import { shallow } from 'enzyme'
+import React from 'react'
+import { App } from '../src/app'
+
+describe('The app layout instance', () => {
+  it('should renders correctly', () => {
```
Comment (herflis) [inline examples/sn-react-memoapp/test/app.test.tsx]:
```suggestion
  it('should be rendered correctly', () => {
```

File: examples/sn-react-memoapp/test/full-screen-loader.test.tsx [hunk 9]
```diff
@@ -0,0 +1,9 @@
+import { shallow } from 'enzyme'
+import React from 'react'
+import { FullScreenLoader } from '../src/components/full-screen-loader'
+
+describe('The full screen loader instance', () => {
+  it('should renders correctly', () => {
```
Comment (herflis) [inline examples/sn-react-memoapp/test/full-screen-loader.test.tsx]:
```suggestion
  it('should be rendered correctly', () => {
```

File: examples/sn-react-memoapp/test/login-form.test.tsx [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+import TextField from '@material-ui/core/TextField'
+import { LoginForm } from '../src/components/login-form'
+
+describe('The login instance', () => {
+  const testprop = {
+    onLogin: jest.fn(),
+    error: 'test',
+  }
+
+  it('should renders correctly', () => {
```
Comment (herflis) [inline examples/sn-react-memoapp/test/login-form.test.tsx]:
```suggestion
  it('should be rendered correctly', () => {
```

File: examples/sn-react-memoapp/test/login-form.test.tsx [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+    expect(shallow(<LoginForm {...testprop} />)).toMatchSnapshot()
+  })
+
+  it('should change username state when user change it', async () => {
```
Comment (herflis) [inline examples/sn-react-memoapp/test/login-form.test.tsx]:
```suggestion
  it('should change username state when user changes it', async () => {
```

File: examples/sn-react-memoapp/test/login-form.test.tsx [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+    expect(
+      wrapper
+        .update()
+        .find(TextField)
+        .first()
+        .props().value,
+    ).toEqual('testusername')
+  })
+
+  it('should change repository url state when user change it', async () => {
```
Comment (herflis) [inline examples/sn-react-memoapp/test/login-form.test.tsx]:
```suggestion
  it('should change repository url state when user changes it', async () => {
```

File: examples/sn-react-memoapp/test/memo-panel.test.tsx [hunk 13]
```diff
@@ -0,0 +1,165 @@
...
+
+    const memoListElement = wrapper
+      .update()
+      .find('div.MuiExpansionPanelSummary-content')
+      .at(0)
+
+    expect(memoListElement.text()).toContain(TestNewMemo.DisplayName)
+  })
+
+  it('should remove memo from list', async () => {
```
Comment (herflis) [inline examples/sn-react-memoapp/test/memo-panel.test.tsx]:
```suggestion
  it('should remove memo from the list', async () => {
```

File: examples/sn-react-memoapp/test/memo-panel.test.tsx [hunk 14]
```diff
@@ -0,0 +1,165 @@
...
+
+    expect(wrapper.find('div.MuiPaper-root')).toHaveLength(4)
+    await act(async () => {
+      yesbtn.simulate('click')
+    })
+
+    expect(wrapper.update().find('div.MuiPaper-root')).toHaveLength(3)
+  })
+
+  it('should edit memo', async () => {
```
Comment (herflis) [inline examples/sn-react-memoapp/test/memo-panel.test.tsx]:
```suggestion
  it('should modify memo', async () => {
```

File: examples/sn-react-memoapp/test/navbar.test.tsx [hunk 15]
```diff
@@ -0,0 +1,23 @@
+import { mount, shallow } from 'enzyme'
+import React from 'react'
+import Button from '@material-ui/core/Button'
+import { RepositoryContext } from '@sensenet/hooks-react'
+import { NavBarComponent } from '../src/components/navbar'
+
+describe('The navbar instance', () => {
+  it('should renders correctly', () => {
```
Comment (herflis) [inline examples/sn-react-memoapp/test/navbar.test.tsx]:
```suggestion
  it('should be rendered correctly', () => {
```

File: examples/sn-react-memoapp/test/navbar.test.tsx [hunk 16]
```diff
@@ -0,0 +1,23 @@
...
+    expect(shallow(<NavBarComponent />)).toMatchSnapshot()
+  })
+
+  it('should logout correctly', () => {
```
Comment (herflis) [inline examples/sn-react-memoapp/test/navbar.test.tsx]:
```suggestion
  it('should logout the user correctly', () => {
```

**Pull request merged**